cluster-operator/src/main/java/io/strimzi/operator/cluster/model/EntityOperator.java

```diff
@@ -336,4 +338,47 @@ public ServiceAccount generateServiceAccount() {
         }
         return super.generateServiceAccount();
     }
+
+    protected static void javaOptions(List<EnvVar> envVars, JvmOptions jvmOptions, List<SystemProperty> javaSystemProperties) {
+        StringBuilder strimziJavaOpts = new StringBuilder();
+        String xms = jvmOptions != null ? jvmOptions.getXms() : null;
+        if (xms != null) {
+            strimziJavaOpts.append("-Xms").append(xms);
+        }
+
+        String xmx = jvmOptions != null ? jvmOptions.getXmx() : null;
+        if (xmx != null) {
+            strimziJavaOpts.append(" -Xmx").append(xmx);
+        }
+
+        Boolean server = jvmOptions != null ? jvmOptions.isServer() : null;
+
+        if (server != null && server) {
+            strimziJavaOpts.append(' ').append(" -server");
+        }
+
+        Map<String, String> xx = jvmOptions != null ? jvmOptions.getXx() : null;
+        if (xx != null) {
+            xx.forEach((k, v) -> {
+                strimziJavaOpts.append(' ').append("-XX:");
+
+                if ("true".equalsIgnoreCase(v))   {
+                    strimziJavaOpts.append("+").append(k);
+                } else if ("false".equalsIgnoreCase(v)) {
+                    strimziJavaOpts.append("-").append(k);
+                } else  {
+                    strimziJavaOpts.append(k).append("=").append(v);
+                }
```

Comment:
The true/false mapping to +/- seems unintuitive to me. Why not have `JvmOptions.Xx` be a list of XX options. You'd prefix each element  with `-XX:` and just pass it through.

I guess I'd also put the code in ModelUtils, alongside `ModelUtils.getJavaSystemPropertiesToString`. In fact having a `jvmOptionsToString` method in ModelUtils which computes `trim` would be better, I think. Then get rid of this method and let the current caller of this method invoke `jvmOptionsToString` and decide how it gets passed to the container (in particular which env var gets set).

Reply:
> The true/false mapping to +/- seems unintuitive to me. Why not have JvmOptions.Xx be a list of XX options. You'd prefix each element with -XX: and just pass it through.

This already exists for all the Kafka based pods. So we cannot easily change it there. And the whole idea of the PR was to have the same configuration in all places not have different stuff in different pods. So I'm not sure this is worth it.

Reply:
Well if that's the API we've already adopted, I guess we're stuck with it :frowning_face:.